crates/common_enums/src/enums.rs

```diff
@@ -1943,55 +1944,51 @@ mod custom_serde {
         where
             D: serde::Deserializer<'a>,
         {
-            u32::deserialize(deserializer).map(Country::from_numeric)
+            u32::deserialize(deserializer).map(|value| Country::from_numeric(value).unwrap())
```

Comment:
```suggestion
            u32::deserialize(deserializer).map(Country::from_numeric).map_err(serde::de::Error::custom)
```

Reply:
This won't work because `Country::from_numeric` returns a Result<Country, _>

expected enum `Result<enums::Country, <D as Deserializer<'a>>::Error>`
found enum `Result<Result<enums::Country, NumericCountryCodeParseError>, _>`

Reply:
use `and_then` instead of map
```suggestion
u32::deserialize(deserializer)
                .and_then(|value| Country::from_numeric(value).map_err(serde::de::Error::custom));
```

Reply:
Actually, we are getting 2 types of errors (NumericParseError and deserialization Error). 
So, they cannot be directly mapped to custom error